src/api.h

```diff
@@ -39,25 +34,325 @@ AWS_CRT_API const char *aws_crt_error_debug_str(int);
+typedef struct _aws_crt_http_headers aws_crt_http_headers;
+AWS_CRT_API aws_crt_http_headers *aws_crt_http_headers_new_from_blob(uint8_t *blob, size_t blob_length);
+AWS_CRT_API void aws_crt_http_headers_release(aws_crt_http_headers *headers);
+AWS_CRT_API void aws_crt_http_headers_to_blob(
+    aws_crt_http_headers *headers,
+    uint8_t **out_blob,
+    size_t *out_blob_length);
+
+typedef struct _aws_crt_http_message aws_crt_http_message;
+AWS_CRT_API aws_crt_http_message *aws_crt_http_message_new_from_blob(uint8_t *blob, size_t blob_length);
+AWS_CRT_API void aws_crt_http_message_release(aws_crt_http_message *message);
+AWS_CRT_API void aws_crt_http_message_to_blob(
+    aws_crt_http_message *message,
+    uint8_t **out_blob,
+    size_t *out_blob_length);
 
 /* Auth */
-AWS_CRT_API struct aws_credentials *aws_crt_credentials_new(
-    const char *access_key_id, const char *secret_access_key,
-    const char *session_token, uint64_t expiration_timepoint_seconds);
-AWS_CRT_API void
-aws_crt_credentials_release(struct aws_credentials *credentials);
-AWS_CRT_API struct aws_byte_cursor aws_crt_credentials_get_access_key_id(
-    const struct aws_credentials *credentials);
-AWS_CRT_API struct aws_byte_cursor aws_crt_credentials_get_secret_access_key(
-    const struct aws_credentials *credentials);
-AWS_CRT_API struct aws_byte_cursor aws_crt_credentials_get_session_token(
-    const struct aws_credentials *credentials);
-AWS_CRT_API uint64_t aws_crt_credentials_get_expiration_timepoint_seconds(
-    const struct aws_credentials *credentials);
+typedef struct aws_credentials aws_crt_credentials;
+typedef struct _aws_crt_credentials_options aws_crt_credentials_options;
+AWS_CRT_API aws_crt_credentials_options *aws_crt_credentials_options_new(void);
+AWS_CRT_API void aws_crt_credentials_options_release(aws_crt_credentials_options *options);
+AWS_CRT_API void aws_crt_credentials_options_set_access_key_id(
+    aws_crt_credentials_options *options,
+    const char *access_key_id);
+AWS_CRT_API void aws_crt_credentials_options_set_secret_access_key(
+    aws_crt_credentials_options *options,
+    const char *secret_access_key);
+AWS_CRT_API void aws_crt_credentials_options_set_session_token(
+    aws_crt_credentials_options *options,
+    const char *session_token);
+AWS_CRT_API void aws_crt_credentials_options_set_expiration_timepoint_seconds(
+    aws_crt_credentials_options *options,
+    uint64_t expiration_timepoint_seconds);
+AWS_CRT_API aws_crt_credentials *aws_crt_credentials_new(aws_crt_credentials_options *options);
+AWS_CRT_API void aws_crt_credentials_release(aws_crt_credentials *credentials);
+AWS_CRT_API struct aws_byte_cursor aws_crt_credentials_get_access_key_id(const aws_crt_credentials *credentials);
+AWS_CRT_API struct aws_byte_cursor aws_crt_credentials_get_secret_access_key(const aws_crt_credentials *credentials);
+AWS_CRT_API struct aws_byte_cursor aws_crt_credentials_get_session_token(const aws_crt_credentials *credentials);
```

Comment:
seeing inconsistency in how strings are returned.
here we're returning `struct aws_byte_cursor`
but `aws_crt_credentials_provider_static_options_get_secret_access_key()` is returning `const char *`

Reply:
Good catch, this is a straight up mistake